<!DOCTYPE html>
<html>
<head>
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>psc_solutions_tutorial02</title>
<link rel="stylesheet" href="https://stackedit.io/res-min/themes/base.css" />
<script type="text/javascript" src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS_HTML"></script>
</head>
<body><div class="container"><p><a href="index.html">Home</a> | <a href="https://github.com/gattil/2016_FiPS_Tutorials/zipball/master">Download repository</a> </p>

<hr>



<h2 id="tutorial-2-evaluating-selection-pressure-with-codon-models"><strong>Tutorial 2</strong>: Evaluating selection pressure with codon models</h2>

<hr>

<p>In this tutorial you will be guided in using PaML to detect natural selection on protein-coding datasets. For some of the following exercises there might be more than one single solution. </p>

<p><em>These exercises were prepared by Maria Anisimova</em></p>

<hr>



<h4 id="1-the-simple-codon-model-with-constant-ω">1. The simple codon model with constant ω</h4>

<p>GOAL: In this exercise we apply the simplest codon model with constant <script type="math/tex" id="MathJax-Element-1"> \omega </script> parameter over tree branches and MSA sites. We run <code>codeml</code> <strong>twice</strong> since we want to compare the log-likelihood of the model when the parameters are optimised on a fixed tree topology (provided with the exercise) and when the branch lengths are re-estimated using the selected substitution model. </p>



<h5 id="execution">Execution</h5>

<p>We are going to use the following dataset: <br>
- <em>bglobin.nuc  (MSA)</em> <br>
- <em>bglobin.tree (Tree)</em></p>

<p><strong>Run 1 – branch lengths fixed</strong>  <br>
We create a new control file for <code>codeml</code> using the template provided in PaML folder. Then, we modify the following settings in order to run the analysis according to the exercise requests:</p>

<pre><code>seqfile = /path/to/bglobin.nuc       * sequence data filename
treefile =/path/to/bglobin.trees     * tree structure file name
outfile = /path/to/output_m0_fixed   * main result file name

seqtype = 1  * 1:codons; 2:AAs; 3:codons--&gt;AAs

model = 0    * models for codons: 0
             *  0:one, 1:b, 2:2 or more dN/dS ratios for branches   

NSsites = 0  * 0:one w;1:neutral;2:selection; 3:discrete;4:freqs;
                * 5:gamma;6:2gamma;7:beta;8:beta&amp;w;9:beta&amp;gamma;
                * 10:beta&amp;gamma+1; 11:beta&amp;normal&gt;1; 12:0&amp;2normal&gt;1;
                * 13:3normal&gt;0

fix_blength = 2  * 0: ignore, -1: random, 1: initial, 2: fixed
</code></pre>

<p><strong>Run 2 – branch lengths re-estimated</strong> </p>

<pre><code>seqfile = /path/to/bglobin.nuc       * sequence data filename
treefile =/path/to/bglobin.trees     * tree structure file name
outfile = /path/to/output_m0_ML      * main result file name

seqtype = 1  * 1:codons; 2:AAs; 3:codons--&gt;AAs

model = 0    * models for codons: 0
             *  0:one, 1:b, 2:2 or more dN/dS ratios for branches   

NSsites = 0  * 0:one w;1:neutral;2:selection; 3:discrete;4:freqs;
                * 5:gamma;6:2gamma;7:beta;8:beta&amp;w;9:beta&amp;gamma;
                * 10:beta&amp;gamma+1; 11:beta&amp;normal&gt;1; 12:0&amp;2normal&gt;1;
                * 13:3normal&gt;0

fix_blength = 1  * 0: ignore, -1: random, 1: initial, 2: fixed
</code></pre>



<h5 id="questions">Questions</h5>

<p><strong>1. Do you observe the codon frequency bias?</strong></p>

<pre><code>Sums of codon usage counts
------------------------------------------------------------------------------
Phe F TTT      65 | Ser S TCT      29 | Tyr Y TAT      12 | Cys C TGT      22
      TTC      68 |       TCC      44 |       TAC      32 |       TGC       6
Leu L TTA       1 |       TCA       4 | *** * TAA       0 | *** * TGA       0
      TTG      15 |       TCG       0 |       TAG       0 | Trp W TGG      43
------------------------------------------------------------------------------
Leu L CTT      13 | Pro P CCT      40 | His H CAT      48 | Arg R CGT       2
      CTC      54 |       CCC      24 |       CAC      78 |       CGC      11
      CTA       6 |       CCA       9 | Gln Q CAA       7 |       CGA       0
      CTG     203 |       CCG       4 |       CAG      65 |       CGG       1
------------------------------------------------------------------------------
Ile I ATT      16 | Thr T ACT      45 | Asn N AAT      49 | Ser S AGT      31
      ATC      28 |       ACC      46 |       AAC      70 |       AGC      19
      ATA       3 |       ACA       4 | Lys K AAA      36 | Arg R AGA       2
Met M ATG      23 |       ACG       1 |       AAG     163 |       AGG      43
------------------------------------------------------------------------------
Val V GTT      62 | Ala A GCT     113 | Asp D GAT      57 | Gly G GGT      49
      GTC      51 |       GCC     132 |       GAC      65 |       GGC     109
      GTA       4 |       GCA      12 | Glu E GAA      40 |       GGA      17
      GTG     147 |       GCG       7 |       GAG      81 |       GGG      17
------------------------------------------------------------------------------
</code></pre>

<p><strong>2. Which position displays the most bias? Why?</strong> </p>

<pre><code>Codon position x base (3x4) table, overall

position  1:    T:0.13930    C:0.23080    A:0.23652    G:0.39338
position  2:    T:0.31005    C:0.20997    A:0.32802    G:0.15196
position  3:    T:0.26675    C:0.34191    A:0.05923    G:0.33211
Average         T:0.23870    C:0.26089    A:0.20792    G:0.29248
</code></pre>

<p><strong>3. What is the ML estimate of the transition-transversion ratio κ?</strong> </p>

<table>
<thead>
<tr>
  <th align="center">fixed branch lengths</th>
  <th align="center">re-estimated branch lengths</th>
</tr>
</thead>
<tbody><tr>
  <td align="center">2.16189</td>
  <td align="center">2.07049</td>
</tr>
</tbody></table>


<p><strong>4. What is the ML estimate of the ω-ratio?</strong> </p>

<table>
<thead>
<tr>
  <th align="center">fixed branch lengths</th>
  <th align="center">re-estimated branch lengths</th>
</tr>
</thead>
<tbody><tr>
  <td align="center">0.20964</td>
  <td align="center">0.23685</td>
</tr>
</tbody></table>


<p><strong>5. How do you interpret these ML estimates?</strong></p>

<hr>



<h4 id="2-branch-models">2. Branch models</h4>

<p>GOAL: In this exercise we estimate the <script type="math/tex" id="MathJax-Element-2"> \omega </script> parameter following several different strategies: a. allowing it to change at each branch on the tree topology, b. fixing it on one single branch (two-categories) c. fixing it on two branches (three-categories).</p>



<h5 id="execution-1">Execution</h5>

<p>For all the following tasks we are going to use the following settings:</p>

<pre><code>seqfile = /path/to/lysozymeSmall.nuc     * sequence data filename
treefile = /path/to/lysozymeSmall.trees  * tree structure file name

seqtype = 1  * 1:codons; 2:AAs; 3:codons--&gt;AAs

fix_blength = 1  * 0: ignore, -1: random, 1: initial, 2: fixed
</code></pre>

<p><strong>a. free-ratio model</strong></p>

<p>In the <em>codeml.ctl</em> file we set the following parameters:</p>

<pre><code>model = 1        * models for codons:
                 *  0:one, 1:b, 2:2 or more dN/dS ratios for branches
</code></pre>

<p><strong>b. two dN/dS ratios</strong></p>

<p>In the <em>codeml.ctl</em> file we set the following parameters:</p>

<pre><code>model = 2        * models for codons:
                 *  0:one, 1:b, 2:2 or more dN/dS ratios for branches
</code></pre>

<p>In the tree file <em>lysozymeSmall.trees</em> we must specify the branch where we think the <script type="math/tex" id="MathJax-Element-3">\omega </script> parameter differs from the rest of the tree.</p>

<blockquote>
  <p>PaML (and codeML) accepts only the following label format at branch level: <code>'#1'</code> where <code>1</code> is an integer greater than 0 </p>
</blockquote>

<pre><code>((Hsa_Human, Hla_gibbon) ,((Cgu/Can_colobus, Pne_langur) '#1', Mmu_rhesus), (Ssc_squirrelM, Cja_marmoset));
</code></pre>

<p>In this case we labeled the branch leading to the clade [colobus, langur]. We are therefore assuming that only this branch will have a value for <script type="math/tex" id="MathJax-Element-4"> \omega </script> that differs from the rest of the tree.</p>

<p><strong>c. three dN/dS ratios</strong></p>

<p>In the <em>codeml.ctl</em> file we set the following parameters:</p>

<pre><code>model = 2        * models for codons:
                 *  0:one, 1:b, 2:2 or more dN/dS ratios for branches
</code></pre>

<p>In the tree file <em>lysozymeSmall.trees</em> we must specify the branches where we think the <script type="math/tex" id="MathJax-Element-5">\omega </script> parameter differs from the rest of the tree.</p>

<blockquote>
  <p>PaML (and codeML) accepts only the following label format at branch level: <code>'#1'</code> where <code>1</code> is an integer greater than 0 </p>
</blockquote>

<pre><code>((Hsa_Human, Hla_gibbon) '#2',((Cgu/Can_colobus, Pne_langur) '#1', Mmu_rhesus), (Ssc_squirrelM, Cja_marmoset));
</code></pre>

<p>In this case we labeled the branch leading to the clade [colobus, langur] and the branch leading to the human clade [Hsa_Human, Hla_gibbon].  We are therefore assuming that only those branches will have values for <script type="math/tex" id="MathJax-Element-6"> \omega </script> that differ from the rest of the tree.</p>



<h5 id="questions-1">Questions</h5>

<p><strong>1. Are results consistent with those presented in ref. (2)?</strong> </p>

<p><strong>2. Based on LRTs, what model fits your data the best (among 2-ratio, 3-ratio and free-ratio models)?</strong>  <br>
<strong>3. What are the degrees of freedoms for each comparison?</strong>  <br>
<strong>4. What can you tell about the evolution of your gene from the ML estimates under this best model?</strong></p>

<hr>



<h4 id="3-site-models">3. Site-models</h4>

<p>GOAL: In this exercise we apply different codon models to a protein-coding dataset in order to evaluate which one fits better our dataset w.r.t. natural selection detection. We evaluate alternative evolutionary hypotheses using nested codon models.</p>



<h5 id="execution-2">Execution</h5>

<p>For all the following tasks we are going to use the following settings:</p>

<pre><code>seqfile = /path/to/bglobin.nuc         * sequence data filename
treefile = /path/to/bglobin.trees      * tree structure file name

seqtype = 1  * 1:codons; 2:AAs; 3:codons--&gt;AAs

fix_blength = 1  * 0: ignore, -1: random, 1: initial, 2: fixed
</code></pre>

<p><strong>Fitting model M1 to M8 in one single run</strong></p>

<pre><code>model = 0
NSsites = 1 2 3 7 8

fix_omega = 0  * 1: omega or omega_1 fixed, 0: estimate
    omega = .4 * initial or fixed omega, for codons or codon-based AAs
</code></pre>

<p><strong>Fitting model M8a</strong></p>

<pre><code>  NSsites = 8
  fix_omega = 1  * 1: omega or omega_1 fixed, 0: estimate
  omega = 1      * initial or fixed omega, for codons or codon-based AAs
</code></pre>



<h5 id="questions-2">Questions</h5>

<p><strong>1. Which models are nested?</strong> </p>

<ul>
<li>M3 vs. M0</li>
<li>M2 vs. M1</li>
<li>M8 vs. M7</li>
<li>M8a vs. M8</li>
</ul>

<p><strong>2. How many degrees of freedom do you use each time to test for significance of the LRT statistic?</strong> </p>

<ul>
<li>M3 vs. M0 :  37-33 = 4 degrees of freedom</li>
<li>M2 vs. M1 : 36-34 = 2 degrees of freedom</li>
<li>M8 vs. M7 : 36-34 = 2 degrees of freedom</li>
<li>M8 vs. M8a: 36-35 = 1 degrees of freedom</li>
</ul>

<p><strong>3. Do your tests suggest positive selection?</strong> </p>

<p><strong>4. Interpret the ML estimates relevant to selective pressure.</strong> </p>

<p><strong>5. If LRTs suggest positive selection, which sites are inferred by the Bayesian approach to be under positive selection (models M2 and M8)?</strong></p>

<p><strong>6. Do NEB and BEB agree on the sites inferred?</strong>  <br>
<strong>7. Compare results from the LRT comparing M7 vs M8 and the LRT comparing M8a vs M8. Are they both significant (or both non-significant)?</strong>  <br>
<strong>8. If they are both significant, does the Bayesian approach predict the same sites?</strong></p>

<hr>

<h4 id="4-branch-site-models-coming-soon">4. Branch-site models <em>(coming soon)</em></h4>

<p>GOAL:</p>



<h5 id="execution-3">Execution</h5>



<h5 id="questions-3">Questions</h5>

<hr>



<h4 id="5-effect-of-the-tree-topology-coming-soon">5. Effect of the tree topology <em>(coming soon)</em></h4>

<p>GOAL:</p>



<h5 id="execution-4">Execution</h5>



<h5 id="questions-4">Questions</h5>

<hr>



<h4 id="6-feeling-adventurous-coming-soon">6. Feeling adventurous? <em>(coming soon)</em></h4>

<p>GOAL:</p>



<h5 id="execution-5">Execution</h5>



<h5 id="questions-5">Questions</h5>

<hr>



<h4 id="references">References</h4>

<ol>
<li><a href="http://abacus.gene.ucl.ac.uk/ziheng/pdf/2005ZhangNielsenYangMBEv22p2472.pdf">Z. Yang, R. Nielsen, N. Goldman, A. Krabbe Pedersen, Codon-Substitution Models for Heterogeneous Selection Pressure at Amino Acid Sites, GENETICS May 1, 2000 vol. 155 no. 1 431-449</a></li>
<li><a href="http://mbe.oxfordjournals.org/content/15/5/568.long">Z Yang, Likelihood ratio tests for detecting positive selection and application to primate lysozyme evolution, Molecular Biology and Evolution, Oxford University Press, 1998</a></li>
<li>Jianzhi Zhang, Rasmus Nielsen, and Ziheng Yang, Evaluation of an Improved Branch-Site Likelihood Method for Detecting Positive Selection at the Molecular Level, Mol Biol Evol (December 2005) 22 (12): 2472-2479 first published online August 17, 2005 <a href="http://mbe.oxfordjournals.org/content/22/12/2472.abstract">doi:10.1093/molbev/msi237</a></li>
<li><a href="http://abacus.gene.ucl.ac.uk/software/pamlDOC.pdf">PaML documentation</a></li>
</ol>

<hr>

<p>Document last updated on 03.01.2016</p>

<hr>

<p>© 2016 Lorenzo Gatti – Applied Computational Genomic Team (ACGT) @ Institute of Applied Simulations (ZHAW) | Wädenswil | Zürich</p></div></body>
</html>