<!DOCTYPE html>
<html>
<head>
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>psc_solutions_tutorial01</title>
<link rel="stylesheet" href="https://stackedit.io/res-min/themes/base.css" />
<script type="text/javascript" src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS_HTML"></script>
</head>
<body><div class="container"><p><a href="index.html">Home</a></p>

<hr>

<h2 id="tutorial-1-inferring-phylogenies-using-maximum-likelihood"><strong>Tutorial 1</strong>: Inferring phylogenies using maximum likelihood</h2>

<p>In this tutorial you will be guided in using PhyML and its extension, CodonPhyML,  to solve common phylogenetic problems. For some of the following exercises there might be more than one single solution.</p>

<hr>



<h3 id="ml-with-dna-data-the-primate-dna-dataset">ML with DNA data (the primate DNA dataset)</h3>



<h4 id="exercise-1">Exercise 1</h4>

<p>GOAL: In this exercise you are asked to run PhyML <strong>twice</strong> in order to compare the effect of estimating nucleotide frequencies from the used dataset vs. optimising them with a maximum likelihood (ML) approach. </p>

<h5 id="execution">Execution</h5>

<p><strong>1. First run</strong> <br>
        - Nucleotide substitution model =  <em>HKY85 + Gamma</em> <br>
        - Estimating <em>transition/transversion</em> ratio (<script type="math/tex" id="MathJax-Element-501">\kappa</script> parameter of HKY85 model) <br>
        - Estimating <em>alpha</em> parameter (remember <script type="math/tex" id="MathJax-Element-502">\alpha = \beta </script> for gamma distributions used in phylogenetics) <br>
        - Estimating <em>nucleotide frequencies</em> <strong>with ML</strong></p>

<p>Here is the list of the parameters to change from the PhyML menu:</p>

<pre><code>[M] ................. Model of nucleotide substitution  HKY85
[F] ................. Optimise equilibrium frequencies  yes
[T] .................... Ts/tv ratio (fixed/estimated)  estimated
[C] ........... Number of substitution rate categories  4
[G] ............. Gamma distributed rates across sites  yes
[A] ... Gamma distribution parameter (fixed/estimated)  estimated
</code></pre>

<p><strong>2. Second run</strong> <br>
        - Nucleotide substitution model =  <em>HKY85 + Gamma</em> <br>
        - Estimating <em>transition/transversion</em> ratio (<script type="math/tex" id="MathJax-Element-503">\kappa</script> parameter of HKY85 model) <br>
        - Estimating <em>alpha</em> parameter (remember <script type="math/tex" id="MathJax-Element-504">\alpha = \beta </script> for gamma distributions used in phylogenetics) <br>
        - Estimating <em>nucleotide frequencies</em> <strong>empirically</strong> from the data</p>

<p>Here is the list of the parameters to change from the PhyML menu:</p>

<pre><code>[M] ................. Model of nucleotide substitution  HKY85
[F] ................. Optimise equilibrium frequencies  no
[T] .................... Ts/tv ratio (fixed/estimated)  estimated
[C] ........... Number of substitution rate categories  4
[G] ............. Gamma distributed rates across sites  yes
[A] ... Gamma distribution parameter (fixed/estimated)  estimated
</code></pre>

<h5 id="questions">Questions</h5>

<ol>
<li><strong>Do you see much difference in the tree?</strong> <br>
     The topology does not change, however we can observe a change in the computed branch length. </li>
<li><p><strong>In the likelihood value (stat file)?</strong> <br>
 The likelihood value of the tree inferred using the nucleotide frequencies estimated empirically from the dataset is lower than the likelihood value of the tree inferred using the nucleotide frequencies estimated via ML. This is due to dataset dimensions (the number of the sequences used). </p>

<pre><code>run 1: -6172.58045
run 2: -6173.49655
</code></pre></li>
<li><p><strong>Which option is best and why do you think so?</strong> <br>
    The best option in this case is to use the ML approach since the ML optimisation technique provides better estimates for the nucleotide frequencies, which, in turn, affect the likelihood of the inferred tree. </p></li>
</ol>

<h4 id="exercise-2">Exercise 2</h4>



<h4 id="exercise-3">Exercise 3</h4>



<h4 id="exercise-4">Exercise 4</h4>



<h4 id="exercise-5">Exercise 5</h4>



<h4 id="exercise-6">Exercise 6</h4>

<hr>



<h3 id="ml-with-protein-data">ML with protein data</h3>



<h3 id="with-primate-dataset">With primate dataset</h3>



<h5 id="exercise-1-1">Exercise 1</h5>



<h5 id="exercise-2-1">Exercise 2</h5>



<h3 id="with-protein-data-sets-from-lerat-et-al-2003">With protein data sets from Lerat et al. (2003)</h3>



<h5 id="exercise-1-2">Exercise 1</h5>



<h5 id="exercise-2-2">Exercise 2</h5>

<hr>

<h4 id="references">References</h4>

<hr>

<p>Document last updated on 31.01.2016</p>

<hr>

<p>© Lorenzo Gatti – Applied Computational Genomic Team (ACGT) @ Institute of Applied Simulations (ZHAW) | Wädenswil | Zürich</p></div></body>
</html>