<!DOCTYPE html>
<html>
<head>
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>psc_solutions_tutorial01</title>
<link rel="stylesheet" href="https://stackedit.io/res-min/themes/base.css" />
<script type="text/javascript" src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS_HTML"></script>
</head>
<body><div class="container"><p><a href="index.html">Home</a></p>

<hr>

<h2 id="tutorial-1-inferring-phylogenies-using-maximum-likelihood"><strong>Tutorial 1</strong>: Inferring phylogenies using maximum likelihood</h2>

<p>In this tutorial you will be guided in using PhyML and its extension, CodonPhyML,  to solve common phylogenetic problems. For some of the following exercises there might be more than one single solution. Herewith I show you my solutions for the exercises prepared by Maria Anisimova</p>

<hr>

<h3 id="ml-with-dna-data-the-primate-dna-dataset">ML with DNA data (the primate DNA dataset)</h3>

<p>Dataset file: <em>primates-nt.phy</em></p>

<h4 id="exercise-1">Exercise 1</h4>

<p>GOAL: In this exercise you are asked to run PhyML <strong>twice</strong> in order to compare the effect of estimating nucleotide frequencies from the used dataset vs. optimising them with a maximum likelihood (ML) approach. </p>

<h5 id="execution">Execution</h5>

<p><strong>1. First run</strong> <br>
        - Nucleotide substitution model =  <em>HKY85 + Gamma</em> <br>
        - Estimating <em>transition/transversion</em> ratio (<script type="math/tex" id="MathJax-Element-501">\kappa</script> parameter of HKY85 model) <br>
        - Estimating <em>alpha</em> parameter (remember <script type="math/tex" id="MathJax-Element-502">\alpha = \beta </script> for gamma distributions used in phylogenetics) <br>
        - Estimating <em>nucleotide frequencies</em> <strong>with ML</strong></p>

<p>Here is the list of the parameters to change from the PhyML menu:</p>

<pre><code>[M] ................. Model of nucleotide substitution  HKY85
[F] ................. Optimise equilibrium frequencies  yes
[T] .................... Ts/tv ratio (fixed/estimated)  estimated
[C] ........... Number of substitution rate categories  4
[G] ............. Gamma distributed rates across sites  yes
[A] ... Gamma distribution parameter (fixed/estimated)  estimated
</code></pre>

<p><strong>2. Second run</strong> <br>
        - Nucleotide substitution model =  <em>HKY85 + Gamma</em> <br>
        - Estimating <em>transition/transversion</em> ratio (<script type="math/tex" id="MathJax-Element-503">\kappa</script> parameter of HKY85 model) <br>
        - Estimating <em>alpha</em> parameter (remember <script type="math/tex" id="MathJax-Element-504">\alpha = \beta </script> for gamma distributions used in phylogenetics) <br>
        - Estimating <em>nucleotide frequencies</em> <strong>empirically</strong> from the data</p>

<p>Here is the list of the parameters to change from the PhyML menu:</p>

<pre><code>[M] ................. Model of nucleotide substitution  HKY85
[F] ................. Optimise equilibrium frequencies  no
[T] .................... Ts/tv ratio (fixed/estimated)  estimated
[C] ........... Number of substitution rate categories  4
[G] ............. Gamma distributed rates across sites  yes
[A] ... Gamma distribution parameter (fixed/estimated)  estimated
</code></pre>

<h5 id="questions">Questions</h5>

<ol>
<li><strong>Do you see much difference in the tree?</strong> <br>
     The topology does not change, however we can observe a change in the computed branch lengths. </li>
</ol>

<table>
<thead>
<tr>
  <th align="center">Nt-frequencies optimised</th>
  <th align="center">Nt-frequencies estimated</th>
</tr>
</thead>
<tbody><tr>
  <td align="center"><img src="./images/ex1_tree1.png" alt="" title=""></td>
  <td align="center"><img src="./images/ex1_tree2.png" alt="" title=""></td>
</tr>
</tbody></table>


<p>2. <strong>In the likelihood value (stat file)?</strong> <br>
     The likelihood value of the tree inferred using the nucleotide frequencies estimated empirically from the dataset is lower than the likelihood value of the tree inferred using the nucleotide frequencies estimated via ML. This is due to dataset dimensions (the number of the sequences used). </p>

<pre><code>    run 1: -6172.58045
    run 2: -6173.49655
</code></pre>

<ol>
<li><strong>Which option is best and why do you think so?</strong> <br>
    The best option in this case is to use the ML approach since the ML optimisation technique provides better estimates for the nucleotide frequencies, which, in turn, affect the likelihood of the inferred tree. This is due to the nucleotide frequencies which are used in the calculation of the likelihood of each site of the alignment (<script type="math/tex" id="MathJax-Element-658">\pi Q = 0</script>, where <script type="math/tex" id="MathJax-Element-659">\pi</script> is the vector of the nucleotide frequencies and <script type="math/tex" id="MathJax-Element-660">Q</script> the rate matrix). </li>
</ol>

<h4 id="exercise-2">Exercise 2</h4>

<p>GOAL: In this exercise you are asked to optimise the tree topology on the substitution parameters obtained using ML performing a tree search (i.e. NNI, SPR, TBR) on the initial tree topology.</p>

<h5 id="execution-1">Execution</h5>

<p><strong>1. Run</strong> <br>
        - Nucleotide substitution model =  <em>HKY85 + Gamma</em> <br>
        - Estimating <em>transition/transversion</em> ratio (<script type="math/tex" id="MathJax-Element-584">\kappa</script> parameter of HKY85 model) <br>
        - Estimating <em>alpha</em> parameter (remember <script type="math/tex" id="MathJax-Element-585">\alpha = \beta </script> for gamma distributions used in phylogenetics) <br>
        - Estimating <em>nucleotide frequencies</em> <strong>with ML</strong> <br>
        - No <em>tree search</em> (tree optimisation)</p>

<p>Here is the list of the parameters to change from the PhyML menu:</p>

<pre><code>[M] ................. Model of nucleotide substitution  HKY85
[F] ................. Optimise equilibrium frequencies  yes
[T] .................... Ts/tv ratio (fixed/estimated)  estimated
[C] ........... Number of substitution rate categories  4
[G] ............. Gamma distributed rates across sites  yes
[A] ... Gamma distribution parameter (fixed/estimated)  estimated
[O] ........................... Optimise tree topoLOGy  no
</code></pre>

<h5 id="questions-1">Questions</h5>

<ol>
<li><strong>Compare the trees obtained with and without tree-search. What do you observe and why?</strong> <br>
The topology inferred without tree search presents a variation in the internal node attribution for the clades (DLangur, Colobus) and (Saki, Titi). </li>
</ol>

<table>
<thead>
<tr>
  <th align="center">With <em>tree-search</em></th>
  <th align="center">Without <em>tree-search</em></th>
</tr>
</thead>
<tbody><tr>
  <td align="center"><img src="./images/ex1_tree1.png" alt="" title=""></td>
  <td align="center"><img src="./images/ex2_tree1.png" alt="" title=""></td>
</tr>
</tbody></table>


<ol>
<li><p><strong>Compare the model estimates with and without tree-search. What do you observe and why?</strong> <br>
The transition/transversion ratio differs slightly between the two runs. As the matter of fact, <script type="math/tex" id="MathJax-Element-674"> k </script> parameter appears to have a higher value when tree-search is not performed. </p></li>
<li><p><strong>Compare the likelihood of the ML and NJ trees. What do you observe and why?</strong> <br>
The likelihood value obtained without tree-search is lower than the value obtain performing the topology optimisation. This value is expected since the substitution parameters are optimised over the first inferred tree topology which did not undergo any refinement. </p>

<pre><code>with tree-search: -6172.58045
without tree-search: -6173.00555
</code></pre></li>
</ol>

<h4 id="exercise-3">Exercise 3</h4>

<table>
<thead>
<tr>
  <th align="center">Model</th>
  <th align="center">Log-likelihood</th>
  <th align="center">Parameters</th>
  <th align="center">AIC</th>
</tr>
</thead>
<tbody><tr>
  <td align="center">JC</td>
  <td align="center">-6379.66383</td>
  <td align="center">38</td>
  <td align="center">12835.32766</td>
</tr>
<tr>
  <td align="center">JC_I</td>
  <td align="center">-6311.41650</td>
  <td align="center">39</td>
  <td align="center">12700.833</td>
</tr>
<tr>
  <td align="center">JC_G</td>
  <td align="center">-6304.38084</td>
  <td align="center">39</td>
  <td align="center">12686.76168</td>
</tr>
<tr>
  <td align="center">JC_G_I</td>
  <td align="center">-6304.35263</td>
  <td align="center">40</td>
  <td align="center">12688.70526</td>
</tr>
<tr>
  <td align="center">HKY</td>
  <td align="center">-6251.43051</td>
  <td align="center">42</td>
  <td align="center">12586.86102</td>
</tr>
<tr>
  <td align="center">HKY_I</td>
  <td align="center">-6180.09689</td>
  <td align="center">43</td>
  <td align="center">12446.19378</td>
</tr>
<tr>
  <td align="center">HKY_G</td>
  <td align="center">-6172.58045</td>
  <td align="center">43</td>
  <td align="center">12431.1609</td>
</tr>
<tr>
  <td align="center">HKY_G_I</td>
  <td align="center">-6172.52896</td>
  <td align="center">44</td>
  <td align="center">12433.05792</td>
</tr>
<tr>
  <td align="center">GTR</td>
  <td align="center">-6241.48788</td>
  <td align="center">46</td>
  <td align="center">12574.97576</td>
</tr>
<tr>
  <td align="center">GTR_I</td>
  <td align="center">-6171.17472</td>
  <td align="center">47</td>
  <td align="center">12436.34944</td>
</tr>
<tr>
  <td align="center">GTR_G</td>
  <td align="center">-6163.87291</td>
  <td align="center">47</td>
  <td align="center">12421.74582</td>
</tr>
<tr>
  <td align="center">GTR_G_I</td>
  <td align="center">-6172.52896</td>
  <td align="center">48</td>
  <td align="center">12441.05792</td>
</tr>
</tbody></table>


<p>The number of parameters are computed in the following way for a rooted tree:</p>

<blockquote>
  <p><script type="math/tex" id="MathJax-Element-814">k = (2* \text{no. taxa} - 2) + \text{no. parameters in substitution model} + \text{extra parameters}</script> <br>
  substitution models:  JC = 0 parameters | HKY = 4 parameters | GTR = 8 parameters <br>
  invariant sites = +1 parameter <br>
  gamma rates = +1 parameter</p>
</blockquote>

<p>After computing the AIC value with </p>

<blockquote>
  <p><script type="math/tex" id="MathJax-Element-815">AIC = 2(k) - \log(L)</script></p>
</blockquote>

<p>we select the model with the lowest AIC value. In this case, <strong>GTR + Gamma</strong></p>

<h4 id="exercise-4">Exercise 4</h4>



<h4 id="exercise-5">Exercise 5</h4>



<h4 id="exercise-6">Exercise 6</h4>

<hr>



<h3 id="ml-with-protein-data">ML with protein data</h3>



<h3 id="with-primate-dataset">With primate dataset</h3>



<h5 id="exercise-1-1">Exercise 1</h5>



<h5 id="exercise-2-1">Exercise 2</h5>



<h3 id="with-protein-data-sets-from-lerat-et-al-2003">With protein data sets from Lerat et al. (2003)</h3>



<h5 id="exercise-1-2">Exercise 1</h5>



<h5 id="exercise-2-2">Exercise 2</h5>

<hr>

<h4 id="references">References</h4>

<hr>

<p>Document last updated on 31.01.2016</p>

<hr>

<p>© 2016 Lorenzo Gatti – Applied Computational Genomic Team (ACGT) @ Institute of Applied Simulations (ZHAW) | Wädenswil | Zürich</p></div></body>
</html>