<!DOCTYPE html>
<html>
<head>
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>psc_solutions_tutorial01</title>
<link rel="stylesheet" href="https://stackedit.io/res-min/themes/base.css" />
<script type="text/javascript" src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS_HTML"></script>
</head>
<body><div class="container"><p><a href="index.html">Home</a> | <a href="https://github.com/gattil/2016_FiPS_Tutorials/zipball/master">Download repository</a> </p>

<hr>



<h2 id="tutorial-1-inferring-phylogenies-using-maximum-likelihood"><strong>Tutorial 1</strong>: Inferring phylogenies using maximum likelihood</h2>

<p>In this tutorial you will be guided in using PhyML and its extension, CodonPhyML,  to solve common phylogenetic problems. For some of the following exercises there might be more than one single solution. </p>

<p><em>These exercises were prepared by Maria Anisimova</em></p>

<hr>



<h3 id="ml-with-dna-data-the-primate-dna-dataset">ML with DNA data (the primate DNA dataset)</h3>

<p>Dataset file: <em>primates-nt.phy</em></p>



<h4 id="exercise-1">Exercise 1</h4>

<p>GOAL: In this exercise you are asked to run PhyML <strong>twice</strong> in order to compare the effect of estimating nucleotide frequencies from the used dataset vs. optimising them with a maximum likelihood (ML) approach. </p>



<h5 id="execution">Execution</h5>

<p><strong>1. First run</strong> <br>
        - Nucleotide substitution model =  <em>HKY85 + Gamma</em> <br>
        - Estimating <em>transition/transversion</em> ratio (<script type="math/tex" id="MathJax-Element-1292">\kappa</script> parameter of HKY85 model) <br>
        - Estimating <em>alpha</em> parameter (remember <script type="math/tex" id="MathJax-Element-1293">\alpha = \beta </script> for gamma distributions used in phylogenetics) <br>
        - Estimating <em>nucleotide frequencies</em> <strong>with ML</strong></p>

<p>Here is the list of the parameters to change:</p>

<p>From 2<sup>nd</sup> menu</p>

<pre><code>[M] ................. Model of nucleotide substitution  HKY85
[F] ................. Optimise equilibrium frequencies  yes
[T] .................... Ts/tv ratio (fixed/estimated)  estimated
[C] ........... Number of substitution rate categories  4
[G] ............. Gamma distributed rates across sites  yes
[A] ... Gamma distribution parameter (fixed/estimated)  estimated
</code></pre>

<p><strong>2. Second run</strong> <br>
        - Nucleotide substitution model =  <em>HKY85 + Gamma</em> <br>
        - Estimating <em>transition/transversion</em> ratio (<script type="math/tex" id="MathJax-Element-1294">\kappa</script> parameter of HKY85 model) <br>
        - Estimating <em>alpha</em> parameter (remember <script type="math/tex" id="MathJax-Element-1295">\alpha = \beta </script> for gamma distributions used in phylogenetics) <br>
        - Estimating <em>nucleotide frequencies</em> <strong>empirically</strong> from the data</p>

<p>Here is the list of the parameters to change:</p>

<p>From 2<sup>nd</sup> menu</p>

<pre><code>[M] ................. Model of nucleotide substitution  HKY85
[F] ................. Optimise equilibrium frequencies  no
[T] .................... Ts/tv ratio (fixed/estimated)  estimated
[C] ........... Number of substitution rate categories  4
[G] ............. Gamma distributed rates across sites  yes
[A] ... Gamma distribution parameter (fixed/estimated)  estimated
</code></pre>



<h5 id="questions">Questions</h5>

<p><strong>1. Do you see much difference in the tree?</strong> <br>
         The topology does not change, however we can observe a change in the computed branch lengths. </p>

<table>
<thead>
<tr>
  <th align="center">Nt-frequencies optimised</th>
  <th align="center">Nt-frequencies estimated</th>
</tr>
</thead>
<tbody><tr>
  <td align="center"><img src="./images/ex1_tree1.png" alt="" title=""></td>
  <td align="center"><img src="./images/ex1_tree2.png" alt="" title=""></td>
</tr>
</tbody></table>


<p><strong>2. In the likelihood value (stat file)?</strong> <br>
     The likelihood value of the tree inferred using the nucleotide frequencies estimated empirically from the dataset is lower than the likelihood value of the tree inferred using the nucleotide frequencies estimated via ML. This is due to dataset dimensions (the number of the sequences used). </p>

<pre><code>    run 1: -6172.58045
    run 2: -6173.49655
</code></pre>

<p><strong>3. Which option is best and why do you think so?</strong> <br>
        The best option in this case is to use the ML approach since the ML optimisation technique provides better estimates for the nucleotide frequencies, which, in turn, affect the likelihood of the inferred tree. This is due to the nucleotide frequencies which are used in the calculation of the likelihood of each site of the alignment (<script type="math/tex" id="MathJax-Element-1296">\pi Q = 0</script>, where <script type="math/tex" id="MathJax-Element-1297">\pi</script> is the vector of the nucleotide frequencies and <script type="math/tex" id="MathJax-Element-1298">Q</script> the rate matrix). </p>



<h4 id="exercise-2">Exercise 2</h4>

<p>GOAL: In this exercise you are asked to optimise the tree topology on the substitution parameters obtained using ML performing a tree search (i.e. NNI, SPR, TBR) on the initial tree topology.</p>



<h5 id="execution-1">Execution</h5>

<p><strong>1. Run</strong> <br>
        - Nucleotide substitution model =  <em>HKY85 + Gamma</em> <br>
        - Estimating <em>transition/transversion</em> ratio (<script type="math/tex" id="MathJax-Element-1299">\kappa</script> parameter of HKY85 model) <br>
        - Estimating <em>alpha</em> parameter (remember <script type="math/tex" id="MathJax-Element-1300">\alpha = \beta </script> for gamma distributions used in phylogenetics) <br>
        - Estimating <em>nucleotide frequencies</em> <strong>with ML</strong> <br>
        - No <em>tree search</em> (tree optimisation)</p>

<p>Here is the list of the parameters to change from the PhyML menu:</p>

<p>From 2<sup>nd</sup> menu</p>

<pre><code>[M] ................. Model of nucleotide substitution  HKY85
[F] ................. Optimise equilibrium frequencies  yes
[T] .................... Ts/tv ratio (fixed/estimated)  estimated
[C] ........... Number of substitution rate categories  4
[G] ............. Gamma distributed rates across sites  yes
[A] ... Gamma distribution parameter (fixed/estimated)  estimated
</code></pre>

<p>From 3<sup>rd</sup> menu</p>

<pre><code>[O] ........................... Optimise tree topoLOGy  no
</code></pre>



<h5 id="questions-1">Questions</h5>

<p><strong>1. Compare the trees obtained with and without tree-search. What do you observe and why?</strong> <br>
The topology inferred without tree search presents a variation in the internal node attribution for the clades (DLangur, Colobus) and (Saki, Titi). </p>

<table>
<thead>
<tr>
  <th align="center">With <em>tree-search</em></th>
  <th align="center">Without <em>tree-search</em></th>
</tr>
</thead>
<tbody><tr>
  <td align="center"><img src="./images/ex1_tree1.png" alt="" title=""></td>
  <td align="center"><img src="./images/ex2_tree1.png" alt="" title=""></td>
</tr>
</tbody></table>


<p><em>* 2.  Compare the model estimates with and without tree-search. What do you observe and why?*</em> <br>
The transition/transversion ratio differs slightly between the two runs. As the matter of fact, <script type="math/tex" id="MathJax-Element-1301"> k </script> parameter appears to have a higher value when tree-search is not performed. </p>

<p><strong>3. Compare the likelihood of the ML and NJ trees. What do you observe and why?</strong> <br>
The likelihood value obtained without tree-search is lower than the value obtain performing the topology optimisation. This value is expected since the substitution parameters are optimised over the first inferred tree topology which did not undergo any refinement. </p>

<pre><code>    with tree-search: -6172.58045
    without tree-search: -6173.00555
</code></pre>



<h4 id="exercise-3">Exercise 3</h4>

<p>GOAL: In this exercise you are asked to compare different substitution models (and their variations) on the same dataset.</p>



<h5 id="execution-2">Execution</h5>

<table>
<thead>
<tr>
  <th align="center">Model</th>
  <th align="center">Log-likelihood</th>
  <th align="center">Parameters</th>
  <th align="center">AIC</th>
</tr>
</thead>
<tbody><tr>
  <td align="center">JC</td>
  <td align="center">-6379.66383</td>
  <td align="center">38</td>
  <td align="center">12835.32766</td>
</tr>
<tr>
  <td align="center">JC_I</td>
  <td align="center">-6311.41650</td>
  <td align="center">39</td>
  <td align="center">12700.833</td>
</tr>
<tr>
  <td align="center">JC_G</td>
  <td align="center">-6304.38084</td>
  <td align="center">39</td>
  <td align="center">12686.76168</td>
</tr>
<tr>
  <td align="center">JC_G_I</td>
  <td align="center">-6304.35263</td>
  <td align="center">40</td>
  <td align="center">12688.70526</td>
</tr>
<tr>
  <td align="center">HKY</td>
  <td align="center">-6251.43051</td>
  <td align="center">42</td>
  <td align="center">12586.86102</td>
</tr>
<tr>
  <td align="center">HKY_I</td>
  <td align="center">-6180.09689</td>
  <td align="center">43</td>
  <td align="center">12446.19378</td>
</tr>
<tr>
  <td align="center">HKY_G</td>
  <td align="center">-6172.58045</td>
  <td align="center">43</td>
  <td align="center">12431.1609</td>
</tr>
<tr>
  <td align="center">HKY_G_I</td>
  <td align="center">-6172.52896</td>
  <td align="center">44</td>
  <td align="center">12433.05792</td>
</tr>
<tr>
  <td align="center">GTR</td>
  <td align="center">-6241.48788</td>
  <td align="center">46</td>
  <td align="center">12574.97576</td>
</tr>
<tr>
  <td align="center">GTR_I</td>
  <td align="center">-6171.17472</td>
  <td align="center">47</td>
  <td align="center">12436.34944</td>
</tr>
<tr>
  <td align="center">GTR_G</td>
  <td align="center">-6163.87291</td>
  <td align="center">47</td>
  <td align="center">12421.74582</td>
</tr>
<tr>
  <td align="center">GTR_G_I</td>
  <td align="center">-6172.52896</td>
  <td align="center">48</td>
  <td align="center">12441.05792</td>
</tr>
</tbody></table>


<p>The number of parameters are computed in the following way for a rooted tree:</p>

<blockquote>
  <p><script type="math/tex" id="MathJax-Element-1302">k = (2* \text{no. taxa} - 2) + \text{no. parameters in substitution model} + \text{extra parameters}</script> <br>
  substitution models:  JC = 0 parameters | HKY = 4 parameters | GTR = 8 parameters <br>
  invariant sites = +1 parameter <br>
  gamma rates = +1 parameter</p>
</blockquote>

<p>We compute the AIC value as follows:</p>

<blockquote>
  <p><script type="math/tex" id="MathJax-Element-1303">AIC = 2(k) - \log(L)</script></p>
</blockquote>



<h5 id="questions-2">Questions</h5>

<p><strong>1. Which model is the best (including HKY+Gamma), based on the AIC criterion?</strong> <br>
We select the model with the lowest AIC value. In this case, <strong>GTR + Gamma</strong></p>

<hr>

<p>This exercise can also be solved using <a href="https://github.com/ddarriba/jmodeltest2">jmodeltest</a> which performs automatically phylogenetic tree reconstructions with every model and extra parameters (i.e. invariant sites,  gamma).</p>



<h4 id="exercise-4">Exercise 4</h4>

<p>GOAL: In this exercise you are asked to compute the branch support using different statistical methods. We are going to evaluate it on the phylogenetic tree evaluated with the following settings:</p>

<pre><code>    . Tree topology search :        NNIs
    . Initial tree:                 BioNJ
    . Model of nucleotides substitution:    GTR
    . Number of taxa:           20
    . Log-likelihood:           -6163.87291
    . Unconstrained likelihood: -5060.29926
    . Parsimony:                741
    . Tree size:                0.54568
    . Discrete gamma model:     Yes
      - Number of categories:           4
      - Gamma shape parameter:      0.565
    . Nucleotides frequencies:
      - f(A)= 0.29130
      - f(C)= 0.21021
      - f(G)= 0.25260
      - f(T)= 0.24589
    . GTR relative rate parameters : 
      A &lt;-&gt; C    1.62195
      A &lt;-&gt; G    4.11843
      A &lt;-&gt; T    0.80648
      C &lt;-&gt; G    1.47358
      C &lt;-&gt; T    3.64931
      G &lt;-&gt; T    1.00000

    . Instantaneous rate matrix : 
      [A---------C---------G---------T------]
      -1.00085   0.21604   0.65916   0.12565  
       0.29937  -1.10379   0.23585   0.56858  
       0.76015   0.19627  -1.11222   0.15580  
       0.14885   0.48607   0.16005  -0.79498  
</code></pre>



<h5 id="execution-3">Execution</h5>

<p>We are going to run PhyML twice: once using the non parametric bootstrap and the second time using the SH-aLRT method. </p>

<p><strong>1. First run</strong></p>

<pre><code>[B] ................ Non parametric bootstrap analysis  yes (100 replicates)
[A] ................ Approximate likelihood ratio test  no
</code></pre>

<p>We perform a bootstrap analysis using 100 replicates (4<sup>th</sup> menu).</p>

<blockquote>
  <p><strong>Bootstrap</strong> analysis assumes that: <br>
  - sites are independently and identically distributed (i.i.d.) <br>
  - Amplifies the biases of the tree inference method <br>
  - Often conservative (&gt;70% is considered strong support, but no guarantee) <br>
  - Slow for large datasets <br>
  - Difficult to interpret</p>
</blockquote>

<p><strong>2. Second run</strong></p>

<pre><code>  [B] ................ Non parametric bootstrap analysis  no
  [A] ................ Approximate likelihood ratio test  yes / aLRT statistics
</code></pre>

<p>We perform an sh-aLRT analysis (4<sup>th</sup> menu).</p>

<blockquote>
  <p><strong>SH-aLRT</strong> method assumes that: <br>
  - the branch being studied provides a significant likelihood gain. <br>
  - per each branch two alternative hypotheses are tested: a null hypothesis where the branch is collapsed and the rest of the topology is maintained vs. an alternative hypothesis where the branch length is kept and the topology is intact.  <br>
  - aLRT is much faster than bootstrap</p>
</blockquote>



<h5 id="questions-3">Questions</h5>

<p><strong>1. Are the results compatible?</strong></p>

<p>Both the methods used to infer the branch support appear to agree on the support provided by long branches, however they do not agree for short ones. These two methods infer branch support using completely different strategies. </p>

<table>
<thead>
<tr>
  <th align="center">With <em>bootstrap</em> support</th>
  <th align="center">With <em>sh-aLRT</em> support</th>
</tr>
</thead>
<tbody><tr>
  <td align="center"><img src="./images/ex4_tree_bootstrap.png" alt="" title=""></td>
  <td align="center"><img src="./images/ex4_tree_sh_aLRT.png" alt="" title=""></td>
</tr>
</tbody></table>




<h4 id="exercise-5">Exercise 5</h4>

<p>For linux/Mac Os X <br>
We assume <em>PhyML</em> binary is included in the user <strong>PATH</strong> variable and it can be called with <em>phyml</em></p>

<p>Positioning in the directory containing the dataset:</p>

<pre><code>cd path/to/tutorial01_phyml
</code></pre>

<p><strong>ex. 1</strong></p>

<pre><code>run1&gt; phyml -i primates-nt.phy -m 'HKY85' -t 'e' -a 'e' -f 'm'
run2&gt; phyml -i primates-nt.phy -m 'HKY85' -t 'e' -a 'e' -f 'e' 
</code></pre>

<p><strong>ex. 2</strong></p>

<pre><code>run1&gt; phyml -i primates-nt.phy -m 'HKY85' -t 'e' -a 'e' -f 'm' -o 'lr'
</code></pre>

<p><strong>ex. 3</strong></p>

<pre><code>JC      &gt; phyml -i primates-nt.phy -m 'JC69' -t 'e' -a 'e' -f 'm' -c 1
JC_I    &gt; phyml -i primates-nt.phy -m 'JC69' -t 'e' -a 'e' -f 'm' -c 1 -v 'e'
JC_G    &gt; phyml -i primates-nt.phy -m 'JC69' -t 'e' -a 'e' -f 'm'
JC_I_G  &gt; phyml -i primates-nt.phy -m 'JC69' -t 'e' -a 'e' -f 'm' -v 'e'

HKY     &gt; phyml -i primates-nt.phy -m 'HKY85' -t 'e' -a 'e' -f 'm' -c 1
HKY_I   &gt; phyml -i primates-nt.phy -m 'HKY85' -t 'e' -a 'e' -f 'm' -c 1 -v 'e'
HKY_G   &gt; phyml -i primates-nt.phy -m 'HKY85' -t 'e' -a 'e' -f 'm'
HKY_I_G &gt; phyml -i primates-nt.phy -m 'HKY85' -t 'e' -a 'e' -f 'm' -v 'e'

GTR     &gt; phyml -i primates-nt.phy -m 'GTR' -t 'e' -a 'e' -f 'm' -c 1
GTR_I   &gt; phyml -i primates-nt.phy -m 'GTR' -t 'e' -a 'e' -f 'm' -c 1 -v 'e'
GTR_G   &gt; phyml -i primates-nt.phy -m 'GTR' -t 'e' -a 'e' -f 'm'
GTR_I_G &gt; phyml -i primates-nt.phy -m 'GTR' -t 'e' -a 'e' -f 'm' -v 'e'
</code></pre>

<p><strong>ex. 4</strong></p>

<pre><code>Bootstrap(100) &gt; phyml -i primates-nt.phy -m 'HKY85' -t 'e' -a 'e' -f 'e' -b 100
SH-aLRT        &gt; phyml -i primates-nt.phy -m 'HKY85' -t 'e' -a 'e' -f 'e' -b -4
</code></pre>



<h4 id="exercise-6">Exercise 6</h4>

<p>GOAL:</p>



<h5 id="execution-4">Execution</h5>

<p><strong>Preprocessing</strong></p>

<p>The dataset should not include any stop codon, therefore we modify the dataset deleting the last 3 columns (the last codon) from the MSA. In order to do so, you can open <em>primates-nt.phy</em> with <strong>AliView</strong> and after selecting the last 3 columns press <em>delete</em> on the keyboard. Save the new file as <em>Phylip</em> with the following name <em>primates-nt-nostop.phy</em>. </p>

<p><strong>For models M0, M0+Gamma and M5</strong></p>

<p>From 1<sup>st</sup> menu</p>

<pre><code>[D] ................. Data type (DNA/AA/CODON/Generic)  CODON
[E] ............................... Matrix exponential  eigenvalues
[H] ... Optimization heuristic for tree reconstruction  original
[O] .................. Parameter optimization strategy  multivariate (BFGS)
[I] ...... Input sequences interleaved (or sequential)  interleaved
[M] ....................... Analyze multiple data sets  no
</code></pre>

<p><strong>M0</strong></p>

<p>From 2<sup>nd</sup> menu</p>

<pre><code>[m] ...... Model type (parametric/empirical/semi-emp.)  parametric
[M] ...................... Model of codon substitution  GY
[i] ..... Initial pairwise distances using codon model  KOSI07
[g] ..................................... Genetic code  Standard
[W] ................. Model for Dn/Ds ratio (M0/M3/M5)  M0
[P] ...................... Equilibrium frequency model  CF3x4
[F] ............................ Origin of frequencies  ML estim.
[T] .................... Ts/Tv ratio (fixed/estimated)  estimated
[D] .................... Dn/Ds ratio (fixed/estimated)  estimated
[V] . Proportion of invariable sites (fixed/estimated)  fixed (p-invar = 0.00)
[R] ....... One category of substitution rate (yes/no)  yes
</code></pre>

<p><strong>M0+Gamma</strong></p>

<p>From 2<sup>nd</sup> menu</p>

<pre><code>[m] ...... Model type (parametric/empirical/semi-emp.)  parametric
[M] ...................... Model of codon substitution  GY
[i] ..... Initial pairwise distances using codon model  KOSI07
[g] ..................................... Genetic code  Standard
[W] ................. Model for Dn/Ds ratio (M0/M3/M5)  M0
[P] ...................... Equilibrium frequency model  CF3x4
[F] ............................ Origin of frequencies  ML estim.
[T] .................... Ts/Tv ratio (fixed/estimated)  estimated
[D] .................... Dn/Ds ratio (fixed/estimated)  estimated
[V] . Proportion of invariable sites (fixed/estimated)  fixed (p-invar = 0.00)
[R] ....... One category of substitution rate (yes/no)  no
[C] ........... Number of substitution rate categories  4
[A] ... Gamma distribution parameter (fixed/estimated)  estimated
[G] .........'Middle' of each rate class (mean/median)  mean
</code></pre>

<p><strong>M5</strong></p>

<pre><code>[m] ...... Model type (parametric/empirical/semi-emp.)  parametric
[M] ...................... Model of codon substitution  GY
[i] ..... Initial pairwise distances using codon model  KOSI07
[g] ..................................... Genetic code  Standard
[W] ................. Model for Dn/Ds ratio (M0/M3/M5)  M5
[w] ................. Number of Dn/Ds ratio categories  3
[P] ...................... Equilibrium frequency model  CF3x4
[F] ............................ Origin of frequencies  ML estim.
[T] .................... Ts/Tv ratio (fixed/estimated)  estimated
[D] .................... Dn/Ds ratio (fixed/estimated)  gamma distribution
[a] .. Gamma distribution alpha/beta (fixed/estimated)  estimated
[G] .........'Middle' of each rate class (mean/median)  mean
[V] . Proportion of invariable sites (fixed/estimated)  fixed (p-invar = 0.00)
</code></pre>

<p><strong>For models LG, WAG, LG+Gamma and WAG+Gamma</strong></p>

<pre><code>[D] ................. Data type (DNA/AA/CODON/Generic)  AA
[d] ............. Convert NT sequence into AA sequence  yes
[G] ..................................... Genetic code  Standard
[I] ...... Input sequences interleaved (or sequential)  interleaved
[M] ....................... Analyze multiple data sets  no
</code></pre>

<p><strong>LG</strong></p>

<pre><code>[M] ................ Model of amino-acids substitution  LG
[F] .. Amino acid frequencies (model/empiric/optimize)  model
[V] . Proportion of invariable sites (fixed/estimated)  fixed (p-invar = 0.00)
[R] ....... One category of substitution rate (yes/no)  yes
</code></pre>

<p><strong>WAG</strong></p>

<pre><code>[M] ................ Model of amino-acids substitution  WAG
[F] .. Amino acid frequencies (model/empiric/optimize)  model
[V] . Proportion of invariable sites (fixed/estimated)  fixed (p-invar = 0.00)
[R] ....... One category of substitution rate (yes/no)  yes
</code></pre>

<p><strong>LG+Gamma</strong></p>

<pre><code>[M] ................ Model of amino-acids substitution  LG
[F] .. Amino acid frequencies (model/empiric/optimize)  model
[V] . Proportion of invariable sites (fixed/estimated)  fixed (p-invar = 0.00)
[R] ....... One category of substitution rate (yes/no)  no
[C] ........... Number of substitution rate categories  4
[A] ... Gamma distribution parameter (fixed/estimated)  estimated
[G] .........'Middle' of each rate class (mean/median)  mean
</code></pre>

<p><strong>WAG+Gamma</strong></p>

<pre><code>[M] ................ Model of amino-acids substitution  WAG
[F] .. Amino acid frequencies (model/empiric/optimize)  model
[V] . Proportion of invariable sites (fixed/estimated)  fixed (p-invar = 0.00)
[R] ....... One category of substitution rate (yes/no)  no
[C] ........... Number of substitution rate categories  4
[A] ... Gamma distribution parameter (fixed/estimated)  estimated
[G] .........'Middle' of each rate class (mean/median)  mean
</code></pre>

<p><strong>GTR+Gamma</strong></p>

<p>From 1<sup>st</sup> menu</p>

<pre><code>[D] ................. Data type (DNA/AA/CODON/Generic)  DNA
[I] ...... Input sequences interleaved (or sequential)  interleaved
[M] ....................... Analyze multiple data sets  no
</code></pre>

<p>From 2<sup>nd</sup> menu</p>

<pre><code>[M] ................. Model of nucleotide substitution  GTR
[F] ................. Optimise equilibrium frequencies  yes
[V] . Proportion of invariable sites (fixed/estimated)  fixed (p-invar = 0.00)
[R] ....... One category of substitution rate (yes/no)  no
[C] ........... Number of substitution rate categories  4
[A] ... Gamma distribution parameter (fixed/estimated)  estimated
[G] .........'Middle' of each rate class (mean/median)  mean
</code></pre>

<p>The number of parameters are computed in the following way for a rooted tree:</p>

<blockquote>
  <p><script type="math/tex" id="MathJax-Element-1304">k = (2* \text{no. taxa} - 2) + \text{no. parameters in substitution model} + \text{extra parameters}</script> <br>
  substitution models:  LG = 0 parameters | WAG = 0 parameters | GTR = 8 parameters | GY = 2 | M0 = 2 | M5 = 2  <br>
  gamma rates = +1 parameter</p>
</blockquote>

<p>We compute the AIC value as follows:</p>

<blockquote>
  <p><script type="math/tex" id="MathJax-Element-1305">AIC = 2(k) - \log(L)</script></p>
</blockquote>

<table>
<thead>
<tr>
  <th align="center">Model</th>
  <th align="center">Log-likelihood</th>
  <th align="center">Parameters</th>
  <th align="center">AIC</th>
</tr>
</thead>
<tbody><tr>
  <td align="center">GTR_G</td>
  <td align="center">-6158.95</td>
  <td align="center">47</td>
  <td align="center">12411.90</td>
</tr>
<tr>
  <td align="center">LG</td>
  <td align="center">-4826.14</td>
  <td align="center">38</td>
  <td align="center">9728.28</td>
</tr>
<tr>
  <td align="center">LG_G</td>
  <td align="center">-4735.17</td>
  <td align="center">39</td>
  <td align="center">9548.34</td>
</tr>
<tr>
  <td align="center">M0</td>
  <td align="center">-6163.53</td>
  <td align="center">40</td>
  <td align="center">12407.06</td>
</tr>
<tr>
  <td align="center">M0_G</td>
  <td align="center">-6075.69</td>
  <td align="center">41</td>
  <td align="center">12233.38</td>
</tr>
<tr>
  <td align="center">M5</td>
  <td align="center">-6057.39</td>
  <td align="center">40</td>
  <td align="center">12194.78</td>
</tr>
<tr>
  <td align="center">WAG</td>
  <td align="center">-4785.42</td>
  <td align="center">38</td>
  <td align="center">9646.84</td>
</tr>
<tr>
  <td align="center">WAG_G</td>
  <td align="center">-4702.97</td>
  <td align="center">39</td>
  <td align="center">9483.94</td>
</tr>
</tbody></table>




<h5 id="questions-4">Questions</h5>

<p><strong>1. Based on AIC, which model fits your dataset best?</strong>  <br>
    WAG + Gamma</p>

<p><strong>2. Are the trees inferred using the best AA and best codon model different?</strong>  <br>
According to the AIC ranking, the best codon model should be <strong>M5</strong>.  Herewith the two trees.</p>

<table>
<thead>
<tr>
  <th align="center">Using <em>codon</em> model</th>
  <th align="center">Using <em>AA</em> model</th>
</tr>
</thead>
<tbody><tr>
  <td align="center"><img src="./images/ex6_tree_M5.png" alt="" title=""></td>
  <td align="center"><img src="./images/ex6_tree_WAG_G.png" alt="" title=""></td>
</tr>
</tbody></table>


<p><strong>3. Are they different to the tree inferred using the DNA model?</strong> <br>
Yes, they are also different to the tree inferred using the GTR+Gamma model. </p>

<hr>



<h3 id="ml-with-protein-data">ML with protein data</h3>



<h4 id="with-primate-dataset">With primate dataset</h4>

<p>GOAL: Testing different amino acids substitution models on a protein dataset <br>
Dataset: <strong>primates-aa.phy</strong></p>



<h5 id="execution-5">Execution</h5>

<p><strong>1. Using PhyML</strong> </p>

<p>We performed the analyses using PhyML and combining the following settings accordingly to the requests:</p>

<pre><code>[M] ................ Model of amino-acids substitution  LG
[F] . Amino acid frequencies (empirical/model defined)  model
[V] . Proportion of invariable sites (fixed/estimated)  fixed (p-invar = 0.00)
[R] ....... One category of substitution rate (yes/no)  yes
</code></pre>

<table>
<thead>
<tr>
  <th align="center">Model</th>
  <th align="center">Log-likelihood</th>
  <th align="center">Parameters</th>
  <th align="center">AIC</th>
</tr>
</thead>
<tbody><tr>
  <td align="center">LG_G</td>
  <td align="center">-5199.88631</td>
  <td align="center">39</td>
  <td align="center">10477.77262</td>
</tr>
<tr>
  <td align="center">LG_G_I</td>
  <td align="center">-5199.55701</td>
  <td align="center">40</td>
  <td align="center">10479.11402</td>
</tr>
<tr>
  <td align="center">LG_G_F</td>
  <td align="center">-5230.44821</td>
  <td align="center">58</td>
  <td align="center">10576.89642</td>
</tr>
<tr>
  <td align="center">LG_G_F_I</td>
  <td align="center">-5230.08104</td>
  <td align="center">59</td>
  <td align="center">10578.16208</td>
</tr>
</tbody></table>


<p><strong>2. Using ProtTest webservice</strong> </p>

<p>You can find the results in the file <em>ex7/prottest_analysis.txt</em></p>



<h5 id="questions-5">Questions</h5>

<p><strong>1. Which is the best model based on AIC criterion? What about using invariant sites?</strong> <br>
We selected the <strong>LG + Gamma</strong> model according to the AIC criterion. Using the invariant sites does not influence the model selection. Adding a proportion of invariant sites increases the number of parameters, therefore the AIC penalises the inferred log-likelihood value. </p>

<p><strong>2. Compare the trees, likelihood and AIC values to those of LG+Gamma.</strong></p>

<table>
<thead>
<tr>
  <th align="right">Value</th>
  <th align="center">Tree estimated with <em>LG+Gamma</em></th>
  <th align="center">Tree estimated with <em>HIVw+G+F</em></th>
</tr>
</thead>
<tbody><tr>
  <td align="right">Log-Likelihood:</td>
  <td align="center">-5199.89</td>
  <td align="center">-5013.50</td>
</tr>
<tr>
  <td align="right">AIC:</td>
  <td align="center">10477.77</td>
  <td align="center">10141.01</td>
</tr>
<tr>
  <td align="right">Tree</td>
  <td align="center"><img src="./images/ex7_BestTree_PhyML.png" alt="" title=""></td>
  <td align="center"><img src="./images/ex7_BestTree_ProTest.png" alt="" title=""></td>
</tr>
</tbody></table>




<h4 id="with-protein-data-sets-from-lerat-et-al-2003">With protein data sets from Lerat et al. (2003)</h4>

<p>GOAL: Inferring a phylogenetic tree on a protein dataset using an appropriate substitution model and detect an event of HGT.</p>



<h5 id="execution-6">Execution</h5>

<p><strong>1. Using PhyML</strong> </p>

<p>We used a selection of candidate substitution models (JTT, LG, WAG) that are suitable for the protein datasets.</p>

<p>dataset: <strong>Prot_biob.phy</strong></p>

<table>
<thead>
<tr>
  <th align="center">Model</th>
  <th align="center">Log-likelihood</th>
  <th align="center">Parameters</th>
  <th align="center">AIC</th>
</tr>
</thead>
<tbody><tr>
  <td align="center">JTT_G_I</td>
  <td align="center">-4593.61928</td>
  <td align="center">40</td>
  <td align="center">9267.23856</td>
</tr>
<tr>
  <td align="center">LG_G_I</td>
  <td align="center">-4546.51747</td>
  <td align="center">40</td>
  <td align="center">9173.03494</td>
</tr>
<tr>
  <td align="center">WAG_G_I</td>
  <td align="center">-4564.41245</td>
  <td align="center">40</td>
  <td align="center">9208.8249</td>
</tr>
</tbody></table>


<p>We ranked the tested models according to the AIC statistics and we selected the <strong>LG+G+I</strong> model which best fits our dataset</p>

<p><strong>2. Using ProtTest webservice</strong> </p>

<p>You can find the results in the file <em>ex8/prottest_analysis.txt</em></p>



<h5 id="questions-6">Questions</h5>

<p><strong>1. What can you tell based on the ML estimates?</strong></p>

<p><strong>2. Display trees</strong></p>

<table>
<thead>
<tr>
  <th align="center">Using <em>codon</em> model</th>
  <th align="center">Using <em>AA</em> model</th>
</tr>
</thead>
<tbody><tr>
  <td align="center"><img src="./images/ex6_tree_M5.png" alt="" title=""></td>
  <td align="center"><img src="./images/ex6_tree_WAG_G.png" alt="" title=""></td>
</tr>
</tbody></table>


<p><strong>3. Are you able to recover the HGT (horizontal gene transfer) published by Lerat et al.?</strong></p>

<hr>



<h4 id="references">References</h4>

<ol>
<li>Darriba D, Taboada GL, Doallo R, Posada D. 2012. jModelTest 2: more models, new heuristics and parallel computing. Nature Methods 9(8), 772. <a href="http://dx.doi.org/10.1038/nmeth.2109">http://dx.doi.org/10.1038/nmeth.2109</a></li>
<li>Guindon S and Gascuel O (2003). A simple, fast and accurate method to estimate large phylogenies by maximum-likelihood”. Systematic Biology 52: 696-704. <a href="http://sysbio.oxfordjournals.org/content/52/5/696.long">DOI: 10.1080/10635150390235520</a></li>
<li><a href="http://darwin.uvigo.es/download/papers/b04.modelPhylHandbook03.pdf">http://darwin.uvigo.es/download/papers/b04.modelPhylHandbook03.pdf</a></li>
<li>Lerat E, Daubin V, Moran NA (2003) From Gene Trees to Organismal Phylogeny in Prokaryotes:The Case of the γ-Proteobacteria. PLoS Biol 1(1): e19. <a href="http://journals.plos.org/plosbiology/article?id=10.1371/journal.pbio.0000019">doi: 10.1371/journal.pbio.0000019</a></li>
</ol>

<hr>

<p>Document last updated on 02.01.2016</p>

<hr>

<p>© 2016 Lorenzo Gatti – Applied Computational Genomic Team (ACGT) @ Institute of Applied Simulations (ZHAW) | Wädenswil | Zürich</p></div></body>
</html>