<!DOCTYPE html>
<html>
<head>
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>psc_solutions_tutorial02</title>
<link rel="stylesheet" href="https://stackedit.io/res-min/themes/base.css" />
<script type="text/javascript" src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS_HTML"></script>
</head>
<body><div class="container"><p><a href="index.html">Home</a> | <a href="https://github.com/gattil/2016_FiPS_Tutorials/zipball/master">Download repository</a> </p>

<hr>



<h2 id="tutorial-2-evaluating-selection-pressure-with-codon-models"><strong>Tutorial 2</strong>: Evaluating selection pressure with codon models</h2>

<hr>

<p>In this tutorial you will be guided in using PaML to detect natural selection on protein-coding datasets. For some of the following exercises there might be more than one single solution. </p>

<p><em>These exercises were prepared by Maria Anisimova</em></p>

<hr>



<h4 id="1-the-simple-codon-model-with-constant-ω">1. The simple codon model with constant ω</h4>

<p>GOAL: In this exercise we apply the simplest codon model with constant $ \omega $ parameter over tree branches and MSA sites. We run <code>codeml</code> <strong>twice</strong> since we want to compare the log-likelihood of the model when the parameters are optimised on a fixed tree topology (provided with the exercise) and when the branch lengths are re-estimated using the selected substitution model. </p>



<h5 id="execution">Execution</h5>

<p>We are going to use the following dataset: <br>
- <em>bglobin.nuc  (MSA)</em> <br>
- <em>bglobin.tree (Tree)</em></p>

<p><strong>Run 1 – branch lengths fixed</strong>  <br>
We create a new control file for <code>codeml</code> using the template provided in PaML folder. Then, we modify the following settings in order to run the analysis according to the exercise requests:</p>

<pre><code>seqfile = /path/to/bglobin.nuc       * sequence data filename
treefile =/path/to/bglobin.trees     * tree structure file name
outfile = /path/to/output_m0_fixed   * main result file name

seqtype = 1  * 1:codons; 2:AAs; 3:codons--&gt;AAs

model = 0    * models for codons: 0
             *  0:one, 1:b, 2:2 or more dN/dS ratios for branches   

NSsites = 0  * 0:one w;1:neutral;2:selection; 3:discrete;4:freqs;
                * 5:gamma;6:2gamma;7:beta;8:beta&amp;w;9:beta&amp;gamma;
                * 10:beta&amp;gamma+1; 11:beta&amp;normal&gt;1; 12:0&amp;2normal&gt;1;
                * 13:3normal&gt;0

fix_blength = 2  * 0: ignore, -1: random, 1: initial, 2: fixed
</code></pre>

<p><strong>Run 2 – branch lengths re-estimated</strong> </p>

<pre><code>seqfile = /path/to/bglobin.nuc       * sequence data filename
treefile =/path/to/bglobin.trees     * tree structure file name
outfile = /path/to/output_m0_ML      * main result file name

seqtype = 1  * 1:codons; 2:AAs; 3:codons--&gt;AAs

model = 0    * models for codons: 0
             *  0:one, 1:b, 2:2 or more dN/dS ratios for branches   

NSsites = 0  * 0:one w;1:neutral;2:selection; 3:discrete;4:freqs;
                * 5:gamma;6:2gamma;7:beta;8:beta&amp;w;9:beta&amp;gamma;
                * 10:beta&amp;gamma+1; 11:beta&amp;normal&gt;1; 12:0&amp;2normal&gt;1;
                * 13:3normal&gt;0

fix_blength = 1  * 0: ignore, -1: random, 1: initial, 2: fixed
</code></pre>



<h5 id="questions">Questions</h5>

<p><strong>1. Do you observe the codon frequency bias?</strong></p>

<pre><code>Sums of codon usage counts
------------------------------------------------------------------------------
Phe F TTT      65 | Ser S TCT      29 | Tyr Y TAT      12 | Cys C TGT      22
      TTC      68 |       TCC      44 |       TAC      32 |       TGC       6
Leu L TTA       1 |       TCA       4 | *** * TAA       0 | *** * TGA       0
      TTG      15 |       TCG       0 |       TAG       0 | Trp W TGG      43
------------------------------------------------------------------------------
Leu L CTT      13 | Pro P CCT      40 | His H CAT      48 | Arg R CGT       2
      CTC      54 |       CCC      24 |       CAC      78 |       CGC      11
      CTA       6 |       CCA       9 | Gln Q CAA       7 |       CGA       0
      CTG     203 |       CCG       4 |       CAG      65 |       CGG       1
------------------------------------------------------------------------------
Ile I ATT      16 | Thr T ACT      45 | Asn N AAT      49 | Ser S AGT      31
      ATC      28 |       ACC      46 |       AAC      70 |       AGC      19
      ATA       3 |       ACA       4 | Lys K AAA      36 | Arg R AGA       2
Met M ATG      23 |       ACG       1 |       AAG     163 |       AGG      43
------------------------------------------------------------------------------
Val V GTT      62 | Ala A GCT     113 | Asp D GAT      57 | Gly G GGT      49
      GTC      51 |       GCC     132 |       GAC      65 |       GGC     109
      GTA       4 |       GCA      12 | Glu E GAA      40 |       GGA      17
      GTG     147 |       GCG       7 |       GAG      81 |       GGG      17
------------------------------------------------------------------------------
</code></pre>

<p><strong>2. Which position displays the most bias? Why?</strong> </p>

<pre><code>Codon position x base (3x4) table, overall

position  1:    T:0.13930    C:0.23080    A:0.23652    G:0.39338
position  2:    T:0.31005    C:0.20997    A:0.32802    G:0.15196
position  3:    T:0.26675    C:0.34191    A:0.05923    G:0.33211
Average         T:0.23870    C:0.26089    A:0.20792    G:0.29248
</code></pre>

<p><strong>3. What is the ML estimate of the transition-transversion ratio κ?</strong> </p>

<table>
<thead>
<tr>
  <th align="center">fixed branch lengths</th>
  <th align="center">re-estimated branch lengths</th>
</tr>
</thead>
<tbody><tr>
  <td align="center">2.16189</td>
  <td align="center">2.07049</td>
</tr>
</tbody></table>


<p><strong>4. What is the ML estimate of the ω-ratio?</strong> </p>

<table>
<thead>
<tr>
  <th align="center">fixed branch lengths</th>
  <th align="center">re-estimated branch lengths</th>
</tr>
</thead>
<tbody><tr>
  <td align="center">0.20964</td>
  <td align="center">0.23685</td>
</tr>
</tbody></table>


<p><strong>5. How do you interpret these ML estimates?</strong></p>

<p>The estimates obtained performing a branch length optimisation are tuned on the codon substitution model selected. This procedure optimises the parameters values together with the topology. However, this does not indicate the correctness of the values inferred since this procedure does not perform tree topology rearrangements.  <br>
One can also avoid this procedure disabling the optimisation of the branch lengths. Then, the codon substitution model parameters are optimised at best on the topology provided. </p>

<hr>

<h4 id="2-branch-models">2. Branch models</h4>

<p>GOAL: In this exercise we estimate the $ \omega $ parameter following several different strategies: a. allowing it to change at each branch on the tree topology, b. fixing it on one single branch (two-categories) c. fixing it on two branches (three-categories).</p>



<h5 id="execution-1">Execution</h5>

<p>For all the following tasks we are going to use the following settings:</p>

<pre><code>seqfile = /path/to/lysozymeSmall.nuc     * sequence data filename
treefile = /path/to/lysozymeSmall.trees  * tree structure file name

seqtype = 1  * 1:codons; 2:AAs; 3:codons--&gt;AAs

fix_blength = 1  * 0: ignore, -1: random, 1: initial, 2: fixed
</code></pre>

<p><strong>a. free-ratio model</strong></p>

<p>In the <em>codeml.ctl</em> file we set the following parameters:</p>

<pre><code>model = 1        * models for codons:
                 *  0:one, 1:b, 2:2 or more dN/dS ratios for branches
</code></pre>

<p><strong>b. two dN/dS ratios</strong></p>

<p>In the <em>codeml.ctl</em> file we set the following parameters:</p>

<pre><code>model = 2        * models for codons:
                 *  0:one, 1:b, 2:2 or more dN/dS ratios for branches
</code></pre>

<p>In the tree file <em>lysozymeSmall.trees</em> we must specify the branch where we think the $\omega $ parameter differs from the rest of the tree.</p>

<blockquote>
  <p>PaML (and codeML) accepts only the following label format at branch level: <code>'#1'</code> where <code>1</code> is an integer greater than 0 </p>
</blockquote>

<pre><code>((Hsa_Human, Hla_gibbon) ,((Cgu/Can_colobus, Pne_langur) '#1', Mmu_rhesus), (Ssc_squirrelM, Cja_marmoset));
</code></pre>

<p>In this case we labeled the branch leading to the clade [colobus, langur]. We are therefore assuming that only this branch will have a value for $ \omega $ that differs from the rest of the tree.</p>

<p><strong>c. three dN/dS ratios</strong></p>

<p>In the <em>codeml.ctl</em> file we set the following parameters:</p>

<pre><code>model = 2        * models for codons:
                 *  0:one, 1:b, 2:2 or more dN/dS ratios for branches
</code></pre>

<p>In the tree file <em>lysozymeSmall.trees</em> we must specify the branches where we think the $\omega $ parameter differs from the rest of the tree.</p>

<blockquote>
  <p>PaML (and codeML) accepts only the following label format at branch level: <code>'#1'</code> where <code>1</code> is an integer greater than 0 </p>
</blockquote>

<pre><code>((Hsa_Human, Hla_gibbon) '#2',((Cgu/Can_colobus, Pne_langur) '#1', Mmu_rhesus), (Ssc_squirrelM, Cja_marmoset));
</code></pre>

<p>In this case we labeled the branch leading to the clade [colobus, langur] and the branch leading to the human clade [Hsa_Human, Hla_gibbon].  We are therefore assuming that only those branches will have values for $ \omega $ that differ from the rest of the tree.</p>



<h5 id="questions-1">Questions</h5>

<p><strong>1. Are results consistent with those presented in ref. (2)?</strong> </p>

<p><strong>2. Based on LRTs, what model fits your data the best (among 2-ratio, 3-ratio and free-ratio models)?</strong> </p>

<p><strong>3. What are the degrees of freedoms for each comparison?</strong> </p>

<p><strong>4. What can you tell about the evolution of your gene from the ML estimates under this best model?</strong></p>

<hr>



<h4 id="3-site-models">3. Site-models</h4>

<p>GOAL: In this exercise we apply different codon models to a protein-coding dataset in order to evaluate which one fits better our dataset w.r.t. natural selection detection. We evaluate alternative evolutionary hypotheses using nested codon models.</p>



<h5 id="execution-2">Execution</h5>

<p>For all the following tasks we are going to use the following settings:</p>

<pre><code>seqfile = /path/to/bglobin.nuc         * sequence data filename
treefile = /path/to/bglobin.trees      * tree structure file name

seqtype = 1  * 1:codons; 2:AAs; 3:codons--&gt;AAs

fix_blength = 1  * 0: ignore, -1: random, 1: initial, 2: fixed
</code></pre>

<p><strong>Fitting model M1 to M8 in one single run</strong></p>

<pre><code>model = 0
NSsites = 1 2 3 7 8

fix_omega = 0  * 1: omega or omega_1 fixed, 0: estimate
    omega = .4 * initial or fixed omega, for codons or codon-based AAs
</code></pre>

<p><strong>Fitting model M8a</strong></p>

<pre><code>  NSsites = 8
  fix_omega = 1  * 1: omega or omega_1 fixed, 0: estimate
  omega = 1      * initial or fixed omega, for codons or codon-based AAs
</code></pre>



<h5 id="questions-2">Questions</h5>

<p><strong>1. Which models are nested?</strong> </p>

<ul>
<li>M3 vs. M0</li>
<li>M2 vs. M1</li>
<li>M8 vs. M7</li>
<li>M8a vs. M8</li>
</ul>

<blockquote>
  <p>We compute the LRT statistic between nested hypothesis using the following relation:</p>
  
  <p>$l_0 = \log[L(H_0)]$ <br>
  $l_1 = \log[L(H_1)]$ <br>
  $LRT = 2\delta = 2 \log\frac{L(H_1)}{L(H_0)} = 2(l_1 - l_0) $</p>
  
  <p>If the $H_0$ hypothesis is correct, then it is asymptotically distributed as the $\chi^2$ distribution with $k$ degrees of freedom ($k$ is equal to the difference in the number of parameters in $H_0$ and $H_1$. On the other hand, I will discard the simplest model in favour of the most complex one, if the LRT value is greater of the value of the $\chi^2$ distribution with $k$ degrees of freedom for the p-value I selected (i.e. p-value = 0.05)</p>
</blockquote>

<table>
<thead>
<tr>
  <th align="center">model</th>
  <th align="center">degrees of freedom</th>
  <th align="left">LRT</th>
  <th>Significant (p-value = 0.01)</th>
</tr>
</thead>
<tbody><tr>
  <td align="center">M3 vs M0</td>
  <td align="center">4</td>
  <td align="left">256.9</td>
  <td>yes, I reject M0</td>
</tr>
<tr>
  <td align="center">M2 vs M1</td>
  <td align="center">2</td>
  <td align="left">7.190</td>
  <td>no, I accept M1</td>
</tr>
<tr>
  <td align="center">M8 vs M7</td>
  <td align="center">2</td>
  <td align="left">22.178</td>
  <td>yes, I reject M7</td>
</tr>
<tr>
  <td align="center">M8 vs M8a</td>
  <td align="center">1</td>
  <td align="left">12.261</td>
  <td>yes, I reject M8a</td>
</tr>
</tbody></table>


<p><strong>2. How many degrees of freedom do you use each time to test for significance of the LRT statistic?</strong> </p>

<ul>
<li>M3 vs. M0 :  37-33 = 4 degrees of freedom</li>
<li>M2 vs. M1 : 36-34 = 2 degrees of freedom</li>
<li>M8 vs. M7 : 36-34 = 2 degrees of freedom</li>
<li>M8 vs. M8a: 36-35 = 1 degrees of freedom</li>
</ul>

<p><strong>3. Do your tests suggest positive selection?</strong> </p>

<p>We tested for positive selection every time we compare the likelihood of two nested models (i.e. M2 vs M1) where one allows for positive selection (it allows $\omega &gt; 1 $ for portion of sites on the MSA), while the other does not ($\omega \lt 1 $).</p>

<p>M2a vs M1a, M8 vs M7, M8 vs M8a model tests can detect positive selection.</p>

<p><strong>4. Interpret the ML estimates relevant to selective pressure.</strong> </p>

<p>Codon models can identify positive selection monitoring the $\omega$ parameter (i.e. on the branches or at each site), therefore we can observe the ML estimates of the omega parameter when inferred under different codon models:</p>

<table>
<thead>
<tr>
  <th align="center">parameter</th>
  <th align="center">M0</th>
  <th align="center">M1</th>
  <th align="center">M2</th>
  <th align="center">M3</th>
  <th align="center">M7</th>
  <th align="center">M8</th>
  <th align="center">M8a</th>
</tr>
</thead>
<tbody><tr>
  <td align="center">$\omega$</td>
  <td align="center">0.23685</td>
  <td align="center">0.12 <br>  1.00</td>
  <td align="center">0.12 <br> 1.00 <br> 2.94</td>
  <td align="center">0.02 <br> 0.30 <br> 1.69</td>
  <td align="center">0.001 <br> 0.008 <br>  0.03 <br> 0.07 <br> 0.13 <br>  0.21 <br> 0.32 <br> 0.45 <br> 0.63 <br> 0.85</td>
  <td align="center">0.002 <br> 0.015 <br> 0.038 <br> 0.070 <br> 0.112 <br> 0.165 <br> 0.233 <br> 0.323 <br> 0.446 <br> 0.651 <br> 2.081</td>
  <td align="center">0.004 <br> 0.022 <br> 0.050 <br> 0.088 <br> 0.139 <br> 0.207 <br> 0.307 <br> 0.491 <br> 1.00</td>
</tr>
<tr>
  <td align="center">proportion</td>
  <td align="center">1.00</td>
  <td align="center">0.77 <br> 0.23</td>
  <td align="center">0.76 <br> 0.20 <br> 0.03</td>
  <td align="center">0.39 <br> 0.54 <br> 0.08</td>
  <td align="center">0.10 <br> 0.10 <br> 0.10 <br> 0.10 <br> 0.10 <br> 0.10 <br> 0.10 <br> 0.10 <br> 0.10 <br> 0.10</td>
  <td align="center">0.09 <br> 0.09 <br> 0.09 <br> 0.09 <br> 0.09 <br> 0.09 <br> 0.09 <br> 0.09 <br> 0.09 <br> 0.09 <br> 0.06</td>
  <td align="center">0.11 <br> 0.11 <br> 0.11 <br> 0.11 <br> 0.11 <br> 0.11 <br> 0.11 <br> 0.11 <br> 0.11</td>
</tr>
</tbody></table>


<p><strong>M0: ConstantOmega</strong>: The $\omega$ ML estimates inferred using the M0 model provide a rough average value distributed over the MSA sites. M0 does not allow for distinct $\omega$ classes, therefore we will not retrieve any information regards positive selection. </p>

<p><strong>M1: NearlyNeutral (2 categories)</strong>: This model allows for two different $\omega$ value classes ($\omega =1 , \omega &lt; 1$).  We observe that the 77% of sites shows an $\omega = 0.12$, while 23% of sites shows an $\omega = 1.00$. If this model is selected by LRT,  one can then discuss about a nearly neutral selection that acted on the dataset. </p>

<p><strong>M2: PositiveSelection (3 categories)</strong>: This model allows for three different classes of $\omega$ (negative selection, neutral selection, positive selection). We observe the following that the majority (76%) of sites are under negative selection, 20% of sites is neutrally selected and only the 3% of sites is positively selected (with an $\omega$ value = 2.94). The LRT test comparing M2 and M1 selected M1 in favour of M2 with a significance cutoff of 1%.</p>

<p><strong>M3: discrete (3 categories)</strong>:  This model estimates the $\omega$ value without forcing it to belong to the conventional classes (i.e. negative, neutral, positive). The ML estimation estimates 3 distinct averaged values for the omega parameter from 3 distinct site proportions. As matter of fact, here we retrieved 2/3 of $\omega$ values which indicate negative selection ($\omega &lt; 1$). The centre of mass of the sites is located in the negative selection region of the $\omega$ value distribution.</p>

<p><strong>M7: beta (10 categories)</strong>: This model infers the $\omega$ parameter values using a Beta distribution with parameters p and q. The ML routine optimises these parameters and cuts the distribution in 10 equal classes. We can observe that all the sites of our MSA are equally distributed across the inferred $\omega$ values, without bias. This model does not account for a specific omega class which can indicate positive selection. </p>

<p><strong>M8: beta&amp;w&gt;1 (11 categories)</strong>: M8 model is a direct extension of M7 model since it accounts for an additional $\omega$ class for positive selected sites. We observe an homogeneous distribution of the sites among the omega classes describing neutral or negative selection, while a small proportion is shown to be under strict positive selection.  From LRT comparison, we select the model M8 over its alternative M7, confirming the presence of a portion of sites under positive selection.</p>

<p><strong>M8a: beta&amp;w=1 (11 categories)</strong>: This variation of the M8 model fixes the upper boundary of the $\omega$ parameter to assume (at maximum) a neutral value $\omega = 1$. Despite we observe an equal distribution of the sites among the $\omega$ classes, the hypothesis assumed by this model is discarded in favour of the M8 model allowing for positive selection.  </p>

<p><strong>5. If LRTs suggest positive selection, which sites are inferred by the Bayesian approach to be under positive selection (models M2 and M8)?</strong></p>

<p><strong>M2 –  discarded</strong></p>

<pre><code>Naive Empirical Bayes (NEB) analysis
Positively selected sites (*: P&gt;95%; **: P&gt;99%)
(amino acids refer to 1st sequence: human)

        Pr(w&gt;1)     post mean +- SE for w

 7 S      0.913         2.772
50 D      0.889         2.725
67 G      0.835         2.620
123 P      0.872         2.693

Bayes Empirical Bayes (BEB) analysis (Yang, Wong &amp; Nielsen 2005. Mol. Biol. Evol. 22:1107-1118)
Positively selected sites (*: P&gt;95%; **: P&gt;99%)
(amino acids refer to 1st sequence: human)

        Pr(w&gt;1)     post mean +- SE for w

 7 S      0.922         2.929 +- 0.956
48 T      0.542         2.037 +- 1.072
50 D      0.905         2.904 +- 0.991
67 G      0.860         2.798 +- 1.039
123 P      0.883         2.818 +- 0.988
</code></pre>

<p><strong>M8 – accepted</strong></p>

<pre><code>Naive Empirical Bayes (NEB) analysis
Positively selected sites (*: P&gt;95%; **: P&gt;99%)
(amino acids refer to 1st sequence: human)

        Pr(w&gt;1)     post mean +- SE for w

 7 S      0.995**       2.074
42 S      0.845         1.856
48 T      0.942         1.997
50 D      0.990**       2.067
54 G      0.781         1.762
67 G      0.988*        2.064
85 T      0.866         1.887
123 P      0.995**       2.075


Bayes Empirical Bayes (BEB) analysis (Yang, Wong &amp; Nielsen 2005. Mol. Biol. Evol. 22:1107-1118)
Positively selected sites (*: P&gt;95%; **: P&gt;99%)
(amino acids refer to 1st sequence: human)

        Pr(w&gt;1)     post mean +- SE for w

 7 S      0.979*        2.442 +- 0.534
42 S      0.539         1.625 +- 0.846
48 T      0.833         2.176 +- 0.753
50 D      0.970*        2.430 +- 0.554
54 G      0.520         1.594 +- 0.870
67 G      0.961*        2.413 +- 0.573
85 T      0.616         1.774 +- 0.860
123 P      0.976*        2.435 +- 0.540
</code></pre>

<p><strong>6. Do NEB and BEB agree on the sites inferred?</strong> </p>

<p>Both the methods agree on the sites inferred, however they compute different significance values. </p>

<p><strong>7. Compare results from the LRT comparing M7 vs M8 and the LRT comparing M8a vs M8. Are they both significant (or both non-significant)?  If they are both significant, does the Bayesian approach predict the same sites?</strong> </p>

<p>The LRT statistics selects M8 over M7 and M8 over M8a. </p>

<hr>



<h4 id="4-branch-site-models-coming-soon">4. Branch-site models <em>(coming soon)</em></h4>

<p>GOAL:</p>



<h5 id="execution-3">Execution</h5>



<h5 id="questions-3">Questions</h5>

<hr>



<h4 id="5-effect-of-the-tree-topology-coming-soon">5. Effect of the tree topology <em>(coming soon)</em></h4>

<p>GOAL:</p>



<h5 id="execution-4">Execution</h5>



<h5 id="questions-4">Questions</h5>

<hr>



<h4 id="6-feeling-adventurous-coming-soon">6. Feeling adventurous? <em>(coming soon)</em></h4>

<p>GOAL:</p>



<h5 id="execution-5">Execution</h5>



<h5 id="questions-5">Questions</h5>

<hr>



<h4 id="references">References</h4>

<ol>
<li><a href="http://abacus.gene.ucl.ac.uk/ziheng/pdf/2005ZhangNielsenYangMBEv22p2472.pdf">Z. Yang, R. Nielsen, N. Goldman, A. Krabbe Pedersen, Codon-Substitution Models for Heterogeneous Selection Pressure at Amino Acid Sites, GENETICS May 1, 2000 vol. 155 no. 1 431-449</a></li>
<li><a href="http://mbe.oxfordjournals.org/content/15/5/568.long">Z Yang, Likelihood ratio tests for detecting positive selection and application to primate lysozyme evolution, Molecular Biology and Evolution, Oxford University Press, 1998</a></li>
<li>Jianzhi Zhang, Rasmus Nielsen, and Ziheng Yang, Evaluation of an Improved Branch-Site Likelihood Method for Detecting Positive Selection at the Molecular Level, Mol Biol Evol (December 2005) 22 (12): 2472-2479 first published online August 17, 2005 <a href="http://mbe.oxfordjournals.org/content/22/12/2472.abstract">doi:10.1093/molbev/msi237</a></li>
<li><a href="http://abacus.gene.ucl.ac.uk/software/pamlDOC.pdf">PaML documentation</a></li>
</ol>

<hr>

<p>Document last updated on 03.01.2016</p>

<hr>

<p>© 2016 Lorenzo Gatti – Applied Computational Genomic Team (ACGT) @ Institute of Applied Simulations (ZHAW) | Wädenswil | Zürich</p></div></body>
</html>